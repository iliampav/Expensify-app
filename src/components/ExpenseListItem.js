import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
 
const ExpenseListItem = ({dispatch, id, description, amount, createdAt}) => (
    //lista dos itens
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>
            {numeral(amount / 100).format('$0,0.00')}
            - 
            {moment(createdAt).format('Do MMMM, YYYY')}
         </p>    u
    </div>
)

export default ExpenseListItem