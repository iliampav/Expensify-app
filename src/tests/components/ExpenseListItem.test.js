import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('Should render a expense item from fixture data', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />)
    expect(wrapper).toMatchSnapshot();
})
