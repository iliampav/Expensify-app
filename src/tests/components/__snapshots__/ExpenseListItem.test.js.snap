// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a expense item from fixture data 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    $1.95
    -
    31st December, 1969
  </p>
      u
</div>
`;
